src/components/02_atoms/Widgets/FileUploadWidget.js

```diff
@@ -50,153 +51,220 @@ const styles = {
   `,
 };
 
-const FileUploadWidget = ({
-  value,
-  label,
-  bundle,
-  onChange,
-  fieldName,
-  inputProps,
-  entityTypeId,
-  required,
-  schema: { properties, maxItems },
-  classes,
-}) => {
-  // If array then allow for multiple uploads.
-  const multiple = properties.data.type === 'array';
-
-  const items = getItemsAsArray(multiple, value.data)
-    // Default schema creates stub entries, which we don't need here.
-    .filter(item => item.id);
-  const length = (items && items.length) || 0;
-  // maxItems is only set if array, so set to 1 as default.
-  const maxItemsCount = multiple ? maxItems || 100000000000 : 1;
-
-  return (
-    <FormControl
-      margin="normal"
-      required={required}
-      classes={classes}
-      fullWidth
-    >
-      <Element>
-        <FormLabel component="legend">{label}</FormLabel>
-        <div
-          className={styles.fullWidth}
-          style={{
-            display:
-              (!multiple && length) || length === maxItemsCount
-                ? 'none'
-                : 'block',
-          }}
-        >
-          <FileUpload
-            bundle={bundle}
-            multiple={multiple}
-            fieldName={fieldName}
-            inputProps={inputProps}
-            entityTypeId={entityTypeId}
-            remainingUploads={maxItemsCount - length}
-            onFileUpload={files => {
-              const newItems = files.reduce((itemsAgg, file) => {
-                const item = {
-                  file: {
-                    type: 'file--file',
-                    url: file.url[0].value,
-                    id: file.uuid[0].value,
-                    filename: file.filename[0].value,
-                  },
-                  meta: { alt: '' },
-                  id: file.uuid[0].value,
-                  type: 'file--file',
-                };
-                return setItemById(multiple, item, itemsAgg);
-              }, items);
-
-              onChange({
-                data: newItems,
-              });
+class FileUploadWidget extends React.Component {
+  state = {
+    selectedItems: null,
+  };
+
+  componentDidMount() {
+    if (!this.state.selectedItems && this.props.value) {
+      const entityTypeId = 'file';
+      const bundle = 'file';
+
+      let ids;
+      if (Array.isArray(this.props.value.data)) {
+        ids = this.props.value.data.map(entity => {
+          return entity.id;
+        });
+      } else {
+        ids = [this.props.value.data.id];
+      }
+      this.fetchEntitites(entityTypeId, bundle, ids).then(({ data: items }) => {
+        this.setState({
+          selectedItems: items.map(({ id, attributes }, index) => ({
+            id,
+            type: 'file--file',
+            [entityTypeId]: attributes,
+            meta: this.props.value.data[index].meta,
```

Comment:
@dawehner Here's an interesting issue. If there is only one entity returned, you can't fetch it by it's index; it's at `this.props.value.data.meta`. You can probably check like, `Array.isArray(this.props.value.data) ? this.props.value.data[index].meta : this.props.value.data.meta`

Reply:
good point!